moderation/punishments.go

```diff
@@ -439,14 +440,25 @@ func RemoveMemberMuteRole(config *Config, id int64, currentRoles []int64, mute M
 
 	newMemberRoles := make([]string, 0, len(currentRoles)+len(config.MuteRemoveRoles))
 
+	gs := bot.State.Guild(true, config.GuildID)
+	gs.RLock()
+	defer gs.RUnlock()
```

Comment:
This would hold the guild lock during a API call, to avoid this create a new function for creating the new member roles where the state is locked and return the new roles from that.

Reply:
Updated the code to fix it!